File: 1. 两数之和.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = {}; // value: index
    for (let i = 0; i < nums.length; i++) {
        const nowValue = nums[i];
        const restValue = target - nowValue;
        if (map.hasOwnProperty(restValue)) {
            return [i, map[restValue]];
        }
        map[nowValue] = i;
    }
    return [];
};

File: 1. 两数之和.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> m;
        for (int i = 0; i < nums.size(); i++) {
            int now = nums[i];
            int rest = target - now;
            if (m.find(rest) != m.end()) {
                return {i, m[rest]};
            }
            m[now] = i;
        }
        return {};
    }
};